### PR TITLE
fix(quantic): update timeframe facet range validations

### PR DESCRIPTION
**Pull request by lbergeron**
https://coveord.atlassian.net/browse/SFINT-4263

After some testing with UX, here is the updated validations for the Timeframe facet. You will find the ADR with details on the actual implementation, but here are the leading principles behind it:

1. Give the end user the opportunity to fill in the range before showing validation errors (i.e., wait for the Apply button to be clicked)
2. The form is optional and therefore input fields become required only when the form is filled. The errors related to required fields should disappear automatically when interacting with other components, or when erasing the field content.
3. The errors related to invalid form input (e.g., invalid format, invalid range) should persist even when interacting with other components. These error messages are still relevant because the range must be corrected to be applied.

[Here is a short video showing the validations](https://share.vidyard.com/watch/RkwQWjfauvVoPjpQYYRVhY?)

### CODE REVIEW

**Review by adroletCoveo — APPROVED** [review]
Thanks for the adjustments Luc!

File: packages/quantic/cypress/integration/facets/timeframe-facet/timeframe-facet.cypress.ts [hunk 1]
```diff
@@ -279,6 +279,30 @@ describe('quantic-timeframe-facet', () => {
...
+          Expect.validationError('Complete this field.');
+
+          Actions.typeEndDate(validRange.end);
+          Actions.submitForm();
+          cy.wait(InterceptAliases.Search);
+
+          Expect.numberOfValidationErrors(0);
+
+          Actions.clearFilter();
+          cy.wait(InterceptAliases.Search);
```
Comment (nathanlb) [inline packages/quantic/cypress/integration/facets/timeframe-facet/timeframe-facet.cypress.ts:303]: is there supposed to be a validation after this?

**Review by nathanlb — APPROVED** [review]
LGTM!

File: packages/quantic/cypress/integration/facets/timeframe-facet/timeframe-facet.cypress.ts (hunk 1 above)
Comment (lbergeron) [reply packages/quantic/cypress/integration/facets/timeframe-facet/timeframe-facet.cypress.ts:303]: No, it is simply there to reset the facet state.

**Comment by nathanlb** [comment]
Now that it works like we want can we also have the range enabled in the exampleSearch?

**Comment by lbergeron** [comment]
> Now that it works like we want can we also have the range enabled in the exampleSearch?

Yes, this PR enables the option.

**Pull request merged**